packages/flame_tiled/lib/src/renderable_tile_map.dart

```diff
@@ -463,16 +474,134 @@ class _RenderableTileLayer extends _RenderableLayer<TileLayer> {
 
     var staggerY = 0.0;
     var staggerX = 0.0;
-    if (_map.orientation == MapOrientation.hexagonal) {
-      // Hexagonal Ponity Tiles move down by a fractional amount.
+    // Hexagonal Ponity Tiles move down by a fractional amount.
+    if (_map.staggerAxis == StaggerAxis.y) {
+      staggerY = size.y * 0.5;
+    } else
+    // Hexagonal Flat Tiles move right by a fractional amount.
+    if (_map.staggerAxis == StaggerAxis.x) {
+      staggerX = size.x * 0.5;
+    }
+
+    for (var ty = 0; ty < tileData.length; ty++) {
+      final tileRow = tileData[ty];
+
+      // Hexagonal Pointy Tiles shift left and right depending on the row
       if (_map.staggerAxis == StaggerAxis.y) {
-        staggerY = _map.tileHeight * 0.75;
+        if ((ty.isOdd && _map.staggerIndex == StaggerIndex.odd) ||
+            (ty.isEven && _map.staggerIndex == StaggerIndex.even)) {
+          staggerX = halfDestinationTile.x;
+        } else {
+          staggerX = 0.0;
+        }
+      }
+
+      // When staggering in the X axis, we need to hold painting of "lower"
+      // tiles (those with staggerY adjustments) otherwise they'll just get
+      // painted over. See the second pass loop after tx.
+      final xSecondPass = <_Transform>[];
+
+      for (var tx = 0; tx < tileRow.length; tx++) {
+        final tileGid = tileRow[tx];
+        if (tileGid.tile == 0) {
+          continue;
+        }
+
+        final tile = _map.tileByGid(tileGid.tile);
```

Comment:
in the next version we'll have to add `!` when getting `tileByGid` because `tiled.dart` has some changes in the next version and this returns `Tile?` now